src/client/Microsoft.Identity.Client/ManagedIdentity/KeyProviders/WindowsCngKeyOperations.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.ManagedIdentity.KeyGuard;
+
+namespace Microsoft.Identity.Client.ManagedIdentity.KeyProviders
+{
+    /// <summary>
```

Comment:
Should we add a minimal retry policy? What about `NTE_DEVICE_NOT_READY` exception (rare with TPM but can occur during initialization)?